airbyte-integrations/connectors/source-harvest/unit_tests/integration/config.py

```diff
@@ -0,0 +1,30 @@
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+
+from datetime import datetime
+from typing import Any, Dict
+
+
+class ConfigBuilder:
+    def __init__(self) -> None:
+        self._config: Dict[str, Any] = {
+            "account_id": "an account id",
+            "replication_start_date": "2021-01-01T00:00:00Z",
+            "credentials": {
+                "api_token": "an api key"
+            }
+        }
+
+    def with_account_id(self, account_id: str) -> "ConfigBuilder":
+        self._config["account_id"] = account_id
+        return self
+
+    def with_replication_start_date(self, replication_start_date: datetime) -> "ConfigBuilder":
+        self._config["start_date"] = replication_start_date.isoformat()[:-13]+"Z"
+        return self
+
+    def with_api_token(self, api_token: str) -> "ConfigBuilder":
+        self._config["credentials"]["api_token"] = api_token
+        return self
+
+    def build(self) -> Dict[str, Any]:
```

Comment:
Where can I read more about `ConfigBuilder` as a concept? For example, I gather that `build()` can hypothetically have dynamic pieces and that's why it's a function — but in this implementation it just returns the underlying dict.

Is `ConfigBuilder` part of the integration tests architecture that we want to see in other connectors? Is it a "standard" class? Should we have a parent class for it that mandates it's behavior then?

Reply:
> Where can I read more about ConfigBuilder as a concept?

Personally, I like builders because they are expressive (if a part of the config is not important to my test, I don't need to configure it) and add a layer of abstraction to avoid breaking change/mass changes

> but in this implementation it just returns the underlying dict.

I think that to be precise, it should be a mapping but eh! https://github.com/airbytehq/airbyte/blob/a99bb8da5bc70c6ffade08044f09e8d3f8813882/airbyte-cdk/python/airbyte_cdk/sources/declarative/spec/spec.py#L22

> Is ConfigBuilder part of the integration tests architecture

Not really as configs are often very specific for connectors